### PR TITLE
Adding ChangeListManager to parse ignored Files in Project and sortin…

### PR DESCRIPTION
**Pull request by ashishrp-aws**
…g through project modules for accurate file parsing.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

- Using ChangeListManager instead of featureDevSessionContext to do ignored file parsing.
- Running through project modules to collect files instead of just collecting all children of projectRoot.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt [hunk 1]
```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
...
+        project.modules.forEach { module ->
+            val changeListManager = ChangeListManager.getInstance(module.project)
+            if (module.guessModuleDir() != null) {
+                stack.push(module.guessModuleDir())
+                while (stack.isNotEmpty()) {
+                    val current = stack.pop()
+
+                    if (!current.isDirectory) {
+                        // Using both featureDevSessionContext and changeListManager to ignore files.
+                        // featureDevSessionContext is used to only ignore files in gitignore in project root and uses regex pattern matching.
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:198]: why wouldn't we just rely on ChangeListManager? it is already gitignore aware

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt [hunk 2]
```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
...
+                        if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) } &&
+                            !changeListManager.isIgnoredFile(current) && !files.contains(current.path)
+                        ) {
+                            if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
+                                break
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:203]: this breaks the inner while loop, but not the modules.forEach loop

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt [hunk 3]
```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
...
 
-        stack.push(projectRoot)
-        while (stack.isNotEmpty()) {
-            val current = stack.pop()
-
-            if (!current.isDirectory) {
-                if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) }) {
-                    if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
-                        break
+        project.modules.forEach { module ->
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:189]: test?

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt (hunk 1 above)
Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:198]: its not checking gitignore files in tests. so thought it'd be prudent to keep both in. i was trying choose to leave featureDevSessionContext in or out.

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt (hunk 3 above)
Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:189]: Wouldn't exisiting create project Payload tests cover this? or do i need add one to confirm modules in project?

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt (hunk 2 above)
Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt:203]: Thanks for the catch.

File: plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererProjectCodeScanTest.kt [hunk 4]
```diff
@@ -378,10 +391,5 @@ class CodeWhispererProjectCodeScanTest : CodeWhispererCodeScanTestBase(PythonCod
         ).virtualFile
         totalSize += testYaml.length
         totalLines += testYaml.toNioPath().toFile().readLines().size
-
-        // Adding gitignore file and gitignore file member for testing.
-        // The tests include the markdown file but not these two files.
-        projectRule.fixture.addFileToProject("/.gitignore", "node_modules\n.idea\n.vscode\n.DS_Store").virtualFile
-        projectRule.fixture.addFileToProject("test.idea", "ref: refs/heads/main")
```
Comment (rli) [inline plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererProjectCodeScanTest.kt:385]: need to figure out tests

**Review by rli — APPROVED** [review]

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererProjectCodeScanTest.kt (hunk 4 above)
Comment (ashishrp-aws) [reply plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererProjectCodeScanTest.kt:385]: Ack. i have been working to write tests including gitignore files.

**Pull request merged**